File: pkg/java/src/main/java/dev/openfga/language/errors/Metadata.java
package dev.openfga.language.errors;

public class Metadata {
    private String symbol;

    private String module;

    public Metadata() {}

    public Metadata(String symbol) {
        this.symbol = symbol;
    }

    public Metadata(String symbol, String module) {
        this.symbol = symbol;
        this.module = module;
    }

    public String getSymbol() {
        return symbol;
    }

    public void setSymbol(String symbol) {
        this.symbol = symbol;
    }

    public String getModule() {
        return module;
    }

    public void setModule(String module) {
        this.module = module;
    }
}

File: pkg/csharp/src/OpenFga.Language/Errors/Metadata.cs
namespace OpenFga.Language.Errors;

/// <summary>
/// Contains basic metadata information for parsing errors.
/// Used to store symbol information associated with parsing errors.
/// </summary>
public class Metadata {
    /// <summary>
    /// Gets or sets the symbol that caused the parsing error.
    /// </summary>
    public string Symbol { get; set; } = string.Empty;

    /// <summary>
    /// Initializes a new instance of the Metadata class.
    /// </summary>
    public Metadata() { }

    /// <summary>
    /// Initializes a new instance of the Metadata class with a symbol.
    /// </summary>
    /// <param name="symbol">The symbol that caused the parsing error</param>
    public Metadata(string symbol) {
        Symbol = symbol;
    }
}